chajda/tsquery/augments.py

```diff
@@ -52,62 +52,185 @@ def augments_gensim(lang, word, config=Config(), n=5):
 
     return words
 
-
+import os
 import fasttext
 import fasttext.util
+from annoy import AnnoyIndex
+
+
+chajda_path = os.path.dirname(__file__)
+working_dir = os.getcwd()
 
 fasttext_models = {}
+annoy_indices = {}
+
+
+def create_annoy_index_if_needed(lang, word='google', n=5):
+    '''
+    Checks whether an AnnoyIndex has already been created for a specified language.
+    Populates an AnnoyIndex  with vectors from the fasttext model that corresponds to language,
+    builds the index, saves it to disk. 
+    The AnnoyIndex is stored in the annoy_indices dictionary.
 
-def augments_fasttext(lang, word, config=Config(), n=5):
+    lang: language passed from augments_fasttext
+    
+    NOTE:
+    Populating an AnnoyIndex with vectors from a fasttext model requires more memory than github actions allows for.
+    Therefore, the doctest has been commented out.
+    # >>> create_annoy_index_if_needed('en') or annoy_indices['en'] != None
+    # True
+    '''
+    # create AnnoyIndex if not already created
+    try:
+        annoy_indices[lang]
+    except KeyError:
+        annoy_indices[lang] = AnnoyIndex(300, 'angular')
+
+        # if annoy index has not been saved for this language yet,
+        # populate annoy index with vectors from corresponding fasttext model
+        try:
+            annoy_indices[lang].load(f"{chajda_path}/annoy/{lang}.ann")
+
+        # OSError occurs if index is not already saved to disk
+        except OSError:
+            # The annoy library uses a random number genertor when building up the trees for an Annoy Index.
+            # In order to ensure that the output is deterministic, we specify the seed value for the random number generator.
+            annoy_indices[lang].set_seed(22)
+
+            # Populate AnnoyIndex with vectors from corresponding fasttext model
```

Comment:
There's an inconsistency here where you use `annoy index` in previous comments and `AnnoyIndex` here.  You should pick one and stay with that everywhere.

I'm assuming that `AnnoyIndex` is the name of the python type, and that's why you used it here?  In general, it's better to use plain english descriptions rather than formal type names unless you are referring to a particular behavior about the type (like that it has a method inside of it or something).  The reason I prefer this is because it's possible that in the future, the annoy library could change the name of the type for some reason, in which case the `AnnoyIndex` comment would become out of date, but the plain English `annoy index` would still be valid.